src/Agent.Sdk/ProcessInvoker.cs

```diff
@@ -805,6 +810,54 @@ private void NixKillProcessTree()
             }
         }
 
+        private void WriteProcessOomScoreAdj(Process proc)
+        {
+            try
+            {
+                if (proc.StartInfo.FileName.Contains("Agent.Worker", StringComparison.OrdinalIgnoreCase) ||
+                    proc.StartInfo.FileName.Contains("Agent.PluginHost", StringComparison.OrdinalIgnoreCase))
+                {
+                    return;
+                }
+                string procFilePath = $"/proc/{proc.Id}/oom_score_adj";
+                // Hosted Ubuntu mounts procfs at /proc, NOP on other platforms
+                if (File.Exists(procFilePath))
+                {
+                    if (proc.StartInfo.Environment.ContainsKey("VSTS_JOB_OOMSCOREADJ"))
+                    {
+                        string userOomScoreAdj = proc.StartInfo.Environment["VSTS_JOB_OOMSCOREADJ"];
+                        File.WriteAllText(procFilePath, userOomScoreAdj);
+                    }
+                    else
+                    {
+                        // Existing score is inherited from parent proc
+                        int oomScoreAdjExisting = int.Parse(File.ReadAllText(procFilePath));
+                        // Agent tends to score around 10, but other procs just need to be higher
+                        // than that. Exact value is arbitrary but we can only increase without sudo.
+                        // Values (up to 1000) make the process more likely to be killed under OOM scenario,
+                        // protecting the agent by extension
+                        if (oomScoreAdjExisting < 0)
+                        {
+                            File.WriteAllText(procFilePath, "0");
+                        }
+                        else if (oomScoreAdjExisting > 900)
+                        {
+                            File.WriteAllText(procFilePath, "1000");
+                        }
+                        else
+                        {
+                            File.WriteAllText(procFilePath, $"{oomScoreAdjExisting + 100}");
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Trace.Verbose($"Failed to update oom_score_adj for {proc.StartInfo.FileName} (PID: {proc.Id}).");
```

Comment:
you might want `Trace.Info`?
should we narrow down the try-catch to only the file operation? i would not expect anything else to throw.

Reply:
I left it as Exception because there are no scenarios where I intend to actually handle anything beyond just logging the io error for now, and since we do not know how people will have their linux machines configured, it might fail in ways I cant predict, eg permissions on the file, the user the agent is running as, etc, making it basically just best effort. So really it seems unnecessary for now since we just log the exception. Can update if you think theres a scenario we should handle better